### DISCUSSION TITLE
Behaviour of void nextTrack()  or void prevTrack() in combination with void playFolderTrack(uint8_t folder, uint8_t track)

### DISCUSSION THREAD

**Discussion by Binary-System** [Makuna/DFMiniMp3#123] [Q&A]
Expected behaviour would be that the current track +1 is played
Example:
```
E:\
+---01
|       001.mp3
|       002.mp3
|       003.mp3
```
If track 2 is played with instance.playFolderTrack(1, 2) 

- instance.nextTrack() should play track 3
- instance.prevTrack() should play track 1

Instead they seem(?) to play the next global track from a random (?) folder on the card.

Is that the correct behaviour?

I consulted the playmp3 example (that expects a fixed track amount) and with the problem from #122 it is impossible to play a folder sequencially without having the track count stored externally on a RFID card for example since I 'cant just add +1 to the track recieved by OnPlayFinished(DfMp3& mp3, DfMp3_PlaySources source, uint16_t track) because it could have been the last one and +1 would trigger a com 5 error because out of bounds

**Comment by Makuna** [comment]
Which actual chip do you have?  MH2024K16SS? YX5200-24SS? Something else?  You will have to look on the chip to see this.  While in general they act similar, there are nuances to each chip and unfortunately, they are placed onto modules labeled and sold as the same. 

I am unsure of the specific expected behavior versus the real behavior.  The documentation on the chips is spotty and often inaccurate. Someone else may chime in with an answer.

**Reply by Binary-System** [reply]
There is something on the chip but i honestly can't read it for sure. It seems like on one smaller one there is TC8002D CM7J8Q1 and on the larger one a large JL or JT logo and below AB???89553 8282. My soldering magining glass is still on the road.

I use the one from here: https://www.leiterkartenpiraten.de/leiterkartenpiraten-player-quick-start-guide/

the even reference you / this repository as reference library for them.

**Comment by Makuna** [comment]
I have heard that depending on the format of the audio codec, sometimes it can act strange.  
Things to try:
Change the compression of your audio to lower bit rate.
Change the codec use for the audio (mp3, wav).
